### ISSUE TITLE
Cannot open, Log states related to TC.

### ISSUE THREAD

**Issue by MonsterGaming54** [SlimeKnights/TinkersConstruct#5042]
### Minecraft Version

1.18.2

### Forge Version

40.1.87

### Mantle Version

1.9.31

### Tinkers' Construct Version

3.5.2.40

### Describe your issue

Game sometimes opens, lets me join a world or server, but immediately crashes if i move my mouse or try walking around. My friends also have the same exact version of the Modpack and are experiencing little to no issues on our server (one guy is crashing occasionally but we suspect it to be his PC).

Registry Object not present: t_construct:scorched_ladder

I saw another issue report saying that it's a fake crash and that it's being caused by something else.

### Crash Report

https://pastebin.com/DTR5N1Eg (Crash Report)

https://pastebin.com/2JfLDM2w (Latest.log)

### Other mods

    [Alex's Mobs (by sbom_xela)](https://www.curseforge.com/minecraft/mc-mods/alexs-mobs)

    [Pipez (by henkelmax)](https://www.curseforge.com/minecraft/mc-mods/pipez)

    [ShetiPhianCore (by ShetiPhian)](https://www.curseforge.com/minecraft/mc-mods/shetiphiancore)

    [Tinkers' Integrations and Tweaks (by wendall911)](https://www.curseforge.com/minecraft/mc-mods/tcintegrations)

    [Waystones (by BlayTheNinth)](https://www.curseforge.com/minecraft/mc-mods/waystones)

    [The Twilight Forest (by Benimatic)](https://www.curseforge.com/minecraft/mc-mods/the-twilight-forest)

    [Sophisticated Core (by P3pp3rF1y)](https://www.curseforge.com/minecraft/mc-mods/sophisticated-core)

    [Mod Name Tooltip (by mezz)](https://www.curseforge.com/minecraft/mc-mods/mod-name-tooltip)

    [Awesome Dungeon The End edition - Forge (by jtl_elisa)](https://www.curseforge.com/minecraft/mc-mods/awesome-dungeon-the-end-forge)

    [JourneyMap (by techbrew)](https://www.curseforge.com/minecraft/mc-mods/journeymap)

    [Flywheel (by jozufozu)](https://www.curseforge.com/minecraft/mc-mods/flywheel)

    [Mekanism Additions (by bradyaidanc)](https://www.curseforge.com/minecraft/mc-mods/mekanism-additions)

    [Corpse (by henkelmax)](https://www.curseforge.com/minecraft/mc-mods/corpse)

    [CodeChicken Lib 1.8.+ (by covers1624)](https://www.curseforge.com/minecraft/mc-mods/codechicken-lib-1-8)

    [Create Crafts & Additions (by MRHminer)](https://www.curseforge.com/minecraft/mc-mods/createaddition)

    [Chisels & Bits - For Forge (by AlgorithmX2)](https://www.curseforge.com/minecraft/mc-mods/chisels-bits)

    [SuperMartijn642's Config Lib (by SuperMartijn642)](https://www.curseforge.com/minecraft/mc-mods/supermartijn642s-config-lib)

    [Applied Energistics 2 (by AlgorithmX2)](https://www.curseforge.com/minecraft/mc-mods/applied-energistics-2)

    [GraveStone Mod (by henkelmax)](https://www.curseforge.com/minecraft/mc-mods/gravestone-mod)

    [Macaw's Fences and Walls (by sketch_macaw)](https://www.curseforge.com/minecraft/mc-mods/macaws-fences-and-walls)

    [Macaw's Furniture (by sketch_macaw)](https://www.curseforge.com/minecraft/mc-mods/macaws-furniture)

    [Architectury API (Fabric/Forge) (by shedaniel)](https://www.curseforge.com/minecraft/mc-mods/architectury-api)

    [Sophisticated Backpacks (by P3pp3rF1y)](https://www.curseforge.com/minecraft/mc-mods/sophisticated-backpacks)

    [Cloth Config API (Fabric/Forge) (by shedaniel)](https://www.curseforge.com/minecraft/mc-mods/cloth-config)

    [Biomes O' Plenty (by Forstride)](https://www.curseforge.com/minecraft/mc-mods/biomes-o-plenty)

    [Plus The End (by praizazdd7724)](https://www.curseforge.com/minecraft/mc-mods/plus-the-end)

    [SuperMartijn642's Core Lib (by SuperMartijn642)](https://www.curseforge.com/minecraft/mc-mods/supermartijn642s-core-lib)

    [Mekanism Tools (by bradyaidanc)](https://www.curseforge.com/minecraft/mc-mods/mekanism-tools)

    [Tinkers' Levelling Addon (by Pyre540)](https://www.curseforge.com/minecraft/mc-mods/tinkers-levelling-addon)

    [TerraBlender (Forge) (by TheAdubbz)](https://www.curseforge.com/minecraft/mc-mods/terrablender)

    [Mouse Tweaks (by YaLTeR)](https://www.curseforge.com/minecraft/mc-mods/mouse-tweaks)

    [Connected Glass (by SuperMartijn642)](https://www.curseforge.com/minecraft/mc-mods/connected-glass)

    [Applied Energistics 2 Wireless Terminals (by Mari_023)](https://www.curseforge.com/minecraft/mc-mods/applied-energistics-2-wireless-terminals)

    [Mob Grinding Utils (by vadis365)](https://www.curseforge.com/minecraft/mc-mods/mob-grinding-utils)

    [Tinkers Reforged (by mrthomas20121)](https://www.curseforge.com/minecraft/mc-mods/tinkers-reforged)

    [Applied Mekanistics (by ramidzkh)](https://www.curseforge.com/minecraft/mc-mods/applied-mekanistics)

    [RFTools Base (by McJty)](https://www.curseforge.com/minecraft/mc-mods/rftools-base)

    [Citadel (by sbom_xela)](https://www.curseforge.com/minecraft/mc-mods/citadel)

    [Just Enough Resources (JER) (by way2muchnoise)](https://www.curseforge.com/minecraft/mc-mods/just-enough-resources-jer)

    [EnderChests (by ShetiPhian)](https://www.curseforge.com/minecraft/mc-mods/enderchests)

    [Portal Gun Totality 1.18.2 [FORGE] (by FishyHard_nespit)](https://www.curseforge.com/minecraft/mc-mods/portal-gun-totality)

    [Create (by simibubi)](https://www.curseforge.com/minecraft/mc-mods/create)

    [Ore Excavation (by Funwayguy)](https://www.curseforge.com/minecraft/mc-mods/ore-excavation)

    [Mekanism (by bradyaidanc)](https://www.curseforge.com/minecraft/mc-mods/mekanism)

    [Just Enough Items (JEI) (by mezz)](https://www.curseforge.com/minecraft/mc-mods/jei)

    [Origins (Forge) (by EdwinMindcraft)](https://www.curseforge.com/minecraft/mc-mods/origins-forge)

    [Balm (Forge Edition) (by BlayTheNinth)](https://www.curseforge.com/minecraft/mc-mods/balm)

    [Charm of Undying (Forge) (by TheIllusiveC4)](https://www.curseforge.com/minecraft/mc-mods/charm-of-undying)

    [spark (by Iucko)](https://www.curseforge.com/minecraft/mc-mods/spark)

    [Cosmetic Armor Reworked (by LainMI)](https://www.curseforge.com/minecraft/mc-mods/cosmetic-armor-reworked)

    [Materialis (by RCXcrafter)](https://www.curseforge.com/minecraft/mc-mods/materialis)

    [Mantle (by mDiyo)](https://www.curseforge.com/minecraft/mc-mods/mantle)

    [Iron Chests (by ProgWML6)](https://www.curseforge.com/minecraft/mc-mods/iron-chests)

    [EnderTanks (by ShetiPhian)](https://www.curseforge.com/minecraft/mc-mods/endertanks)

    [NetherPortalFix (by BlayTheNinth)](https://www.curseforge.com/minecraft/mc-mods/netherportalfix)

    [Neat (by Vazkii)](https://www.curseforge.com/minecraft/mc-mods/neat)

    [Not Enough Wands (by romelo333)](https://www.curseforge.com/minecraft/mc-mods/not-enough-wands)

    [Caelus API (Forge) (by TheIllusiveC4)](https://www.curseforge.com/minecraft/mc-mods/caelus)

    [Mekanism Generators (by bradyaidanc)](https://www.curseforge.com/minecraft/mc-mods/mekanism-generators)

    [Simply Light (by Flanks255)](https://www.curseforge.com/minecraft/mc-mods/simply-light)

    [Ender Storage 1.8.+ (by covers1624)](https://www.curseforge.com/minecraft/mc-mods/ender-storage-1-8)

    [Curios API (Forge) (by TheIllusiveC4)](https://www.curseforge.com/minecraft/mc-mods/curios)

    [Light Overlay (Rift/Forge/Fabric) (by shedaniel)](https://www.curseforge.com/minecraft/mc-mods/light-overlay)

    [Tinkers Construct (by mDiyo)](https://www.curseforge.com/minecraft/mc-mods/tinkers-construct)

    [McJtyLib (by McJty)](https://www.curseforge.com/minecraft/mc-mods/mcjtylib)

    [OpenBlocks Elevator (by vsngarcia)](https://www.curseforge.com/minecraft/mc-mods/openblocks-elevator)

    [Tesseract (by SuperMartijn642)](https://www.curseforge.com/minecraft/mc-mods/tesseract)

    [Enlightened End (by MuffinsQw)](https://www.curseforge.com/minecraft/mc-mods/enlightened-end)

    [Library Ferret - Forge (by jtl_elisa)](https://www.curseforge.com/minecraft/mc-mods/library-ferret-forge)

### Tried reproducing with just Tinkers?

Yes

### Performance Enchancers

None of the above

### Searched for known issues?

Searched open issues

**Comment by KnightMiner** [comment]
I am getting really annoyed by this crash. Some mod is breaking things, like seriously breaking things. No clue who, but it keeps getting reported to us. The crash is literally impossible unless someone changed how forge works.

Narrow down the list of mods needed to reproduce this, get me 5 or fewer ideally.

**Comment by MonsterGaming54** [comment]
I believe it's Open Blocks Elevator and Gravestones that are giving me the problem. I'm in the process of troubleshooting it now.

**Comment by MonsterGaming54** [comment]
It's those two and mekanism additions, confirmed.

**Comment by KnightMiner** [comment]
Please be a bit more specific. This crash no longer happens when you remove those mods? Is it one or both of them?

Would be good to open an issue report on their tracker

**Comment by MonsterGaming54** [comment]
From what i can tell, they're all having individual issues with tinkers. I turn them off one by one (curseforge disable), and my game crashes unless all 3 are off. I will be opening issue reports on the respective trackers.

**Comment by KnightMiner** [comment]
When you do, link the issues here so we can track them

**Comment by KnightMiner** [comment]
Turns out KingLemming recently fixed a similar issue in CoFH core. In his case, the issue was caused by a race condition when multiple of the CoFH mods registered objects at the same time, as Forge loads mods in parallel.

In Tinkers case, while *we* only initialize our deferred registers from a single thread, its possible that other mods are accessing some of our base modules (notably, to fetch creative tab instances in addons) causing them to class load on a different thread. I am not 100% certain that is the issue as race conditions tend to be annoying, but supposing it is I know how to protect against that issue. Will likely require a Mantle update if I want to avoid excessive code duplication.

**Comment by KnightMiner** [comment]
I pushed a fix in 3.6.1.85 that may fix this, requires a Mantle update as well. Not sure if it will work, but as its the closest I have come to a cause for this issue, will close this.

If it happens again and you are on 3.6.1.85, leave a comment with the new crash report.
